src/react/ui-elements/Veeam/VeeamConfiguration.tsx

```diff
@@ -0,0 +1,205 @@
+              render={({ field: { onChange, value } }) => {
+                return (
+                  <Select
+                    id="version"
+                    onChange={(value) => {
+                      onChange(value);
+                    }}
+                    value={value}
+                  >
+                    <Select.Option key="Veeam11" value={'Veeam11'}>
+                      Veeam 11
+                    </Select.Option>
+                    <Select.Option key="Veeam12" value={'Veeam12'}>
+                      Veeam 12
+                    </Select.Option>
+                  </Select>
+                );
+              }}
+            ></Controller>
+          }
+        ></FormGroup>
+        <FormGroup
+          id="name"
+          label="Repository name"
+          direction="vertical"
+          required={true}
+          labelHelpTooltip="TODO"
+          error={errors.name?.message ?? ''}
+          content={
+            <Input
+              id="name"
+              type="text"
+              autoComplete="off"
+              placeholder="Veeam-repository name"
+              {...register('name')}
+            />
+          }
+        ></FormGroup>
+        <FormGroup
+          id="enableImmutableBackup"
+          label="Immutable backup"
+          direction="vertical"
+          help="Enables permanent, unchangeable backups of objects in bucket."
+          helpErrorPosition="bottom"
+          labelHelpTooltip="TODO"
+          content={
+            <Controller
+              name="enableImmutableBackup"
+              control={control}
+              render={({ field: { value, onChange } }) => {
+                return (
+                  <Toggle
+                    id="enableImmutableBackup"
+                    name="enableImmutableBackup"
+                    toggle={value}
+                    label={value ? 'Active' : 'Inactive'}
+                    onChange={(value) => {
+                      onChange(value);
+                    }}
+                  />
+                );
+              }}
+            ></Controller>
+          }
+        ></FormGroup>
+        <FormGroup
+          id="capacity"
+          label="Repository capacity"
+          direction="vertical"
+          error={errors.capacity?.message ?? ''}
+          helpErrorPosition="bottom"
+          labelHelpTooltip="TODO"
+          content={
+            <Stack direction="horizontal">
+              <Controller
+                name="capacity"
+                control={control}
+                render={({ field: { value, onChange } }) => {
+                  return (
+                    <Input
+                      id="capacity"
+                      type="number"
+                      size="1/3"
+                      min={1}
+                      max={1023}
+                      value={value}
+                      onChange={(value) => {
+                        onChange(value);
+                      }}
```

Comment:
nit: I don't think  controller wrapper is necessary for this input component, we can probably spread register instead
```suggestion
                      onChange={onChange}
```